lib/saml20.js

```diff
@@ -97,10 +130,15 @@ exports.create = function(options, callback) {
       // </saml:Attribute>
       var attributeElement = doc.createElementNS(NAMESPACE, 'saml:Attribute');
       attributeElement.setAttribute('Name', prop);
+
+      if (options.includeAttributeNameFormat){
+        attributeElement.setAttribute('NameFormat', getNameFormat(prop));        
+      }
+
       var values = options.attributes[prop] instanceof Array ? options.attributes[prop] : [options.attributes[prop]];
       values.forEach(function (value) {
         var valueElement = doc.createElementNS(NAMESPACE, 'saml:AttributeValue');
-        valueElement.setAttribute('xsi:type', 'xs:anyType');
+        valueElement.setAttribute('xsi:type', options.typedAttributes ? getAttributeType(value) : 'xs:anyType');
```

Comment:
are we handling the case where `value` is undefined? (I don't know if it's possible)